.gitattributes

```diff
@@ -3,3 +3,14 @@ Dockerfile* linguist-language=Dockerfile
 vendor.mod linguist-language=Go-Module
 vendor.sum linguist-language=Go-Checksums
 *.bat text eol=crlf
+*.sh text eol=lf
+
+*.env -text
+
+*.png binary
+*.ico binary
+*.gif binary
+*.jpg binary
+*.sum binary
+*.db binary
+*.tar binary
```

Comment:
these are the binary files I could find in the repository using `grep -rIL .`
Some files such as `.golden` are also reported as binary, but have no content, so I guess it shouldn't be a problem?

Reply:
another option might be to do something like below and let git decide on how to treat the file?
```
testdata/** text=auto
docs/** text=auto
vendor/** text=auto
```

Reply:
Yeah, tricky one;

- vendor: "+1" we should likely never touch anything there
- `testdata/` yes, these are fixtures, so some files may have intentional line-endings
- `docs/` hm... not sure; I think we produce those, and it's probably fine to enforce a specific line-ending?

ISTR there was some "smart" handling in git itself for certain known types (like images), but ... it's been a long time since I looked (and since I worked on Windows, so more often ran into such things) 🤔

Reply:
Some interesting comments in some of those files though; https://github.com/gitattributes/gitattributes/blob/master/.gitattributes#L1-L4

```
# THIS IS ONLY FOR THE gitattributes REPOSITORY.
# Handle line endings automatically for files detected as text
# and leave all files detected as binary untouched.
*               text=auto
```

So wondering if the png and other files are possibly due to some old git version being used (and git should otherwise automatically detect those?)

Reply:
We could opt for more selective enforcement instead of the current approach, like so

```
scripts/** text=auto eol=lf
*.go -text diff=golang
*.bat text=auto eol=crlf
*.sh text=auto eol=lf
```

Reply:
this would probably allow git to automatically handle binary files